File: architect/02week/2-4-2/ASProj/app/src/main/java/org/devio/as/proj/main/MainActivity.java
package org.devio.as.proj.main;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import org.devio.as.proj.common.ui.component.HiBaseActivity;
import org.devio.as.proj.main.R;
import org.devio.as.proj.main.logic.MainActivityLogic;

public class MainActivity extends HiBaseActivity implements MainActivityLogic.ActivityProvider {

    private MainActivityLogic activityLogic;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        activityLogic = new MainActivityLogic(this, savedInstanceState);
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        activityLogic.onSaveInstanceState(outState);
    }
}

File: architect/01week/1-6-2/hi-library/app/src/main/java/org/devio/hi/library/app/MainActivity.kt
package org.devio.hi.library.app

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import org.devio.hi.library.app.demo.log.HiLogDemoActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClick(view: View?) {
        when(view!!.id) {
            R.id.tv_hi_log -> {
                startActivity(Intent(this, HiLogDemoActivity::class.java))
            }
        }
    }

}
